chainer/initializer.py

```diff
@@ -3,6 +3,30 @@
 
 class Initializer(object):
 
+    dtype = numpy.float32
+
+    def __init__(self, dtype=None):
```

Comment:
It seems that `self.dtype` is used only in `generate_array` method. So, I wonder if we really need `dtype` in the constructor.
At least to me, current interface looks confusing. Suppose that we specify `dtype=np.float32` in the constructor while we call `__call__` method with an array of `dtype=np.float64`. In this situation, it is not clear if the initialized array's `dtype` is `np.float32` or `np.float64`. Actually, it will depend on the implementation of each inherited class.
I suggest we remove `dtype` argument here, and make `generate_array` have the `dtype` argument instead. (Sorry, I should have said this earlier...)